README.md

```diff
@@ -76,18 +116,207 @@ load(
+    "codechecker_config"
+)
+```
+
+Create a CodeChecker configuration file e.g. `config.json` (see example [here](https://github.com/Szelethus/codechecker_bazel/blob/readme_update/test/config.json)) and parse it using `codechecker_config()`. 
+
+```python
+codechecker_config(
+    name = "your_codechecker_config",
+    config_file = ":config.json"
+)
+```
+
+Alternatively, you can assemble a CodeChecker configuration without a config file using the rule:
+
+```python
+codechecker_config(
+    name = "your_codechecker_config",
+    analyze = [
+        "--enable=bugprone-dangling-handle",
+        "--enable=bugprone-fold-init-type",
+        "--enable=misc-non-copyable-objects",
+        "--report-hash=context-free-v2",
+    ]
+)
+```
+
+You can now configure your `codechecker`, `codechecker_suite`, `codechecker_test` and `code_checker_test` targets using the above configuration:
 
 ```python
 codechecker_test(
     name = "your_codechecker_rule_name",
+    config = "your_codechecker_config",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+### Per-file CodeChecker analysis: `code_checker_test()`
+> [!IMPORTANT]
+> The rule is still in prototype status and is subject to changes without notice.
+> You are free to experiment and report issues however!
+
+Instead of a single CodeChecker call, the bazel rule `code_checker_test()` invokes
+[`CodeChecker analyze`](https://github.com/Ericsson/codechecker/blob/master/docs/analyzer/user_guide.md#analyze)
+_for each_ translation unit in the targets to analyze. The rule is intended to be
+able to enable incremental analyses and dispatching  analysis jobs to remote build
+agents.
+
+To use `code_checker_test()` include it to your BUILD file:
+
+```python
+load(
+    "@bazel_codechecker//src:code_checker.bzl",
+    "code_checker_test",
+)
+```
+
+Create a `code_checker_test()` target by passing other targets you'd like CodeChecker to analyze:
+
+```python
+code_checker_test(
+    name = "your_code_checker_rule_name",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+```bash
+bazel build ://your_code_checker_rule_name
+```
+
+The analysis results can be found and stored/parsed similarly to [`codechecker_test`](README.md#standard-codechecker-invocation-codechecker_test)
+
+## CodeChecker independent rules
+
+The following rules are _not_ using CodeChecker.
+
+### Clang-tidy: `clang_tidy_aspect` and `clang_tidy_test()`
+
+The Bazel rule `clang_tidy_test` runs clang-tidy natively without CodeChecker. To use it, add the following to your BUILD file:
+
+```python
+load(
+    "@bazel_codechecker//src:clang.bzl",
+    "clang_tidy_test",
+)
+
+clang_tidy_test(
+    name = "your_rule_name",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+The Bazel aspect `clang_tidy_aspect` uses `clang_tidy_test` under the hood, but can be invoked from the command line by passing the following parameter to Bazel build/test: `--aspects @bazel_codechecker//src:clang.bzl%clang_tidy_aspect`.
```

Comment:
I guess vice-versa - all rules use aspect under the hood.

Reply:
My understanding is that its still the only one that is exposed externally, isn't it?